src/iadmin.cpp

```diff
@@ -1624,8 +1625,14 @@ doCommand( char *cmdToken[], rodsArguments_t* _rodsArgs = 0 ) {
         return 0;
     }
     if ( strcmp( cmdToken[0], "rmgroup" ) == 0 ) {
-        generalAdmin( 0, "rm", "user", cmdToken[1],
-                      myEnv.rodsZone, "", "", "", "", "", "" );
+        if(irods::to_version(Conn->svrVersion->relVersion) > irods::version{4, 3, 1}) {
+            generalAdmin( 0, "rm", "group", cmdToken[1],
+                          myEnv.rodsZone, "", "", "", "", "", "" );
+        }
+        else {
+            generalAdmin( 0, "rm", "user", cmdToken[1],
+                          myEnv.rodsZone, "", "", "", "", "", "" );
+        }
```

Comment:
Consider isolating only the things that are changing to the if-ladder.

For example:
```cpp
const char* entity_type = (/* version > 4.3.1 */) ? "group" : "user";
generalAdmin(0, "rm", entity_type, ...);
```

With your current implementation, it's possible for the calls to desync.